internal/pkg/cli/task_run.go

```diff
@@ -134,6 +163,117 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = "./Dockerfile"
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = "latest"
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResource(); err != nil {
+		return err
+	}
+
+	//if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		uri, err := o.pushToECRRepo()
+		if err != nil {
+			return err
+		}
+		o.image = fmt.Sprintf(fmtImageURL, uri, o.imageTag)
+
+		// update image to stack
+		if err := o.deployTaskResource(); err != nil {
+			return err
+		}
+	}
+
+	// TODO: kick off task
+	return nil
+}
+
+func (o *runTaskOpts) getNetworkConfig() error {
```

Comment:
Do we need VPC in the network config?

Reply:
Not really! I only need subnets and security groups to associate the task with